src/console_implementation/game_loop_functions.cpp

```diff
@@ -1,151 +1,197 @@
 #include "main_header.hpp"
 
+bool exit_value = false;  //condition of game_loops (game_loop with render, update and input_loop) working
 
 bool gameloop()
 {
     unsigned int gameplay_window_height;
     unsigned int gameplay_window_width;
 
+    //map setup
     std::string map_name = "level_1_map.txt";
-    std::string map_layout = map_converter(map_name, &gameplay_window_height, &gameplay_window_width); //converting file and getting single string, height and width of gamepolay window
+    std::tuple map_layout = map_converter(map_name); //converting file and getting single string, height and width of gamepolay window
 
-    std::shared_ptr<Ilevel> starting_level = std::make_shared<Tlevel>(map_layout, gameplay_window_height, gameplay_window_width);
-    starting_level -> set_starting_position(Tposition(20,20));
+    std::string map_layout_converted = std::get<0>(map_layout);
+    gameplay_window_height = std::get<1>(map_layout);
+    gameplay_window_width = std::get<2>(map_layout);
```

Comment:
Using c++17 syntax could be even simpler
```suggestion
std::string map_name = "level_1_map.txt";
const auto [map_layout, gameplay_window_height, gameplay_window_width] = map_converter(map_name);
```